### PR TITLE
UIOR-1224: fix currency and amount in a printed order mismatch

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIOR-1224](https://folio-org.atlassian.net/browse/UIOR-1224) - fix currency and amount in a printed order mismatch
## Approach

https://github.com/folio-org/ui-orders/assets/116072773/057efd73-08cf-4c68-8341-d3cd9510d705

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]
order estimated price is already calculated

File: src/PrintOrder/utils.js [hunk 1]
```diff
@@ -64,3 +64,30 @@ export const getOrderPrintData = async (ky, order = {}) => {
...
+  } = poLine;
+
+  let currentExchangeRate = exchangeRate;
+
+  if (currency !== systemCurrency && !exchangeRate) {
+    currentExchangeRate = await getCurrencyRate(ky, currency, systemCurrency);
+  }
+
+  return {
+    totalItems: quantityPhysical + quantityElectronic,
```
Comment (NikitaSedyx) [inline src/PrintOrder/utils.js:90]: how is it connected to price?

File: src/PrintOrder/PrintOrder.js [hunk 2]
```diff
@@ -35,7 +41,9 @@ export const PrintOrderComponent = ({ mutator, order, orderLine, onCancel }) =>
         ? { lines: await getExportData(mutator, linesToPrint, [order], intl) }
         : await getOrderPrintData(ky, order);
 
-      setPrintableOrder(hydrateOrderToPrint({
+      setPrintableOrder(await hydrateOrderToPrint({
+        ky,
```
Comment (NikitaSedyx) [inline src/PrintOrder/PrintOrder.js:45]: don't put ky to hydrateOrderToPrint, introduce hook that will return price based on order lines amount

File: src/PrintOrder/utils.js (hunk 1 above)
Comment (alisher-epam) [reply src/PrintOrder/utils.js:90]: Not sure. I think you worked on this implementation. I just moved the line to utils from here https://github.com/folio-org/ui-orders/pull/1560/files/001714803ece1b7662f328a5721c8b0b106fc734#diff-5f2ed363915b466f2a51ba819555fc6565621677eae5c8a15824e5f9c05d9b97L17-L22

File: src/PrintOrder/PrintOrder.js (hunk 2 above)
Comment (alisher-epam) [reply src/PrintOrder/PrintOrder.js:45]: Sure. Let me think about it.

File: src/PrintOrder/utils.js (hunk 1 above)
Comment (NikitaSedyx) [reply src/PrintOrder/utils.js:90]: but you moved to price specific util, how quantity is connected to price?

File: src/PrintOrder/PrintOrder.js (hunk 2 above)
Comment (alisher-epam) [reply src/PrintOrder/PrintOrder.js]: updated

File: src/PrintOrder/hydrateOrderToPrint.js [hunk 3]
```diff
@@ -14,12 +14,7 @@ export function hydrateOrderToPrint({ order }) {
     ?.find(({ isPrimary }) => isPrimary);
 
   if (order.lines?.length === 1) {
-    const [{ poLineEstimatedPrice, quantityPhysical = 0, quantityElectronic = 0 }] = order.lines;
-
-    totals = {
-      totalItems: quantityPhysical + quantityElectronic,
```
Comment (NikitaSedyx) [inline src/PrintOrder/hydrateOrderToPrint.js:20]: why did you remove totalItems?

File: src/PrintOrder/PrintOrder.js [hunk 4]
```diff
@@ -35,7 +40,8 @@ export const PrintOrderComponent = ({ mutator, order, orderLine, onCancel }) =>
         ? { lines: await getExportData(mutator, linesToPrint, [order], intl) }
         : await getOrderPrintData(ky, order);
 
-      setPrintableOrder(hydrateOrderToPrint({
+      setPrintableOrder(await hydrateOrderToPrint({
+        getPOLineTotalEstimatedPrice,
```
[the remark this answers is not in the document]
Comment (alisher-epam) [reply src/PrintOrder/PrintOrder.js]: updated

File: src/PrintOrder/hydrateOrderToPrint.js (hunk 3 above)
Comment (alisher-epam) [reply src/PrintOrder/hydrateOrderToPrint.js:20]: updated

File: src/PrintOrder/hooks/usePOLineTotalEstimatedPrice/usePOLineTotalEstimatedPrice.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { useOkapiKy, useStripes } from '@folio/stripes/core';
```
Comment (usavkov-epam) [inline src/PrintOrder/hooks/usePOLineTotalEstimatedPrice/usePOLineTotalEstimatedPrice.js:1]: line breaks

File: src/PrintOrder/hooks/usePOLineTotalEstimatedPrice/usePOLineTotalEstimatedPrice.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    let currentExchangeRate = exchangeRate;
+
+    if (currency !== systemCurrency && !exchangeRate) {
+      currentExchangeRate = await getCurrencyRate(currency);
+    }
+
+    return {
+      totalItems: quantityPhysical + quantityElectronic,
+      totalEstimatedPrice: currentExchangeRate ? (poLineEstimatedPrice * currentExchangeRate) : poLineEstimatedPrice,
```
Comment (usavkov-epam) [inline src/PrintOrder/hooks/usePOLineTotalEstimatedPrice/usePOLineTotalEstimatedPrice.js:37]: situation: API fails on exchange rate request [(ref)](https://github.com/folio-org/ui-orders/pull/1560/files#diff-4dd1fefca3f6f23c2a5fe8cf83888b94e2ce562199c0d1988932d0de8f85854bR15)
- `currentExchangeRate = await getCurrencyRate(currency); // currentExchangeRate === {}`  [(ref)](https://github.com/folio-org/ui-orders/pull/1560/files#diff-4dd1fefca3f6f23c2a5fe8cf83888b94e2ce562199c0d1988932d0de8f85854bR32)
- `.. (poLineEstimatedPrice * currentExchangeRate) ..` ⇒ `(3 * {})` => `NaN` [(ref)](https://github.com/folio-org/ui-orders/pull/1560/files#diff-4dd1fefca3f6f23c2a5fe8cf83888b94e2ce562199c0d1988932d0de8f85854bR32)

or I missed smth?

Comment (alisher-epam) [reply src/PrintOrder/hooks/usePOLineTotalEstimatedPrice/usePOLineTotalEstimatedPrice.js:40]: updated

**Review by usavkov-epam — APPROVED** [review]

**Comment by alisher-epam** [comment]
@NikitaSedyx can you review it please?

**Review by NikitaSedyx — APPROVED** [review]
